src/mrinufft/operators/interfaces/torchkbnufft.py

```diff
@@ -0,0 +1,248 @@
+from mrinufft._utils import proper_trajectory, get_array_module
+import numpy as np
+
+TORCH_AVAILABLE = True
+try:
+    import torchkbnufft as tkbn
+    import torch
+
+except ImportError:
+    TORCH_AVAILABLE = False
+
+CUPY_AVAILABLE = True
+try:
+    import cupy as cp
+except ImportError:
+    CUPY_AVAILABLE = False
+
+
+class MRITorchKbNufft(FourierOperatorBase):
+    """MRI Transform Operator using Torch NUFFT.
+
+    Parameters
+    ----------
+    samples: Tensor
+        The samples location of shape ``Nsamples x N_dimensions``.
+        It should be C-contiguous.
+    shape: tuple
+        Shape of the image space.
+    n_coils: int
+        Number of coils.
+    density: bool or Tensor
+       Density compensation support.
+        - If a Tensor, it will be used for the density.
+        - If True, the density compensation will be automatically estimated,
+          using the fixed point method.
+        - If False, density compensation will not be used.
+    smaps: Tensor
+    squeeze_dims: bool, default True
+        If True, will try to remove the singleton dimension for batch and coils.
+    """
+
+    backend = "torchkbnufft"
+    available = TORCH_AVAILABLE
+    autograd_available = False
+
+    def __init__(
+        self,
+        samples,
+        shape,
+        density=False,
+        n_coils=1,
+        n_batchs=1,
+        n_trans=1,
+        smaps=None,
+        eps=1e-6,
+        squeeze_dims=True,
+    ):
+        super().__init__()
+
+        self.shape = shape
+        self.n_coils = n_coils
+        self.eps = eps
+        self.squeeze_dims = squeeze_dims
+        self.n_batchs = n_batchs
+        self.dtype = None
+
+        self._tkb_op = tkbn.KbNufft(im_size=self.shape)
+        self._tkb_adj_op = tkbn.KbNufftAdjoint(im_size=self.shape)
+
+        if isinstance(samples, torch.Tensor):
+            samples = samples.numpy()
+        samples = proper_trajectory(
+            samples.astype(np.float32, copy=False), normalize="pi"
+        )
+        self.samples = torch.tensor(samples)
+
+        self.compute_density(density)
+
+        self.compute_smaps(smaps)
+        if isinstance(smaps, np.ndarray) or isinstance(smaps, cp.ndarray):
+            self.smaps = torch.tensor(smaps)
+
+    @with_torch
+    def op(self, data, ksp=None):
+        """Forward operation.
+
+        Parameters
+        ----------
+        data: Tensor
+
+        Returns
+        -------
+        Tensor: Non-uniform Fourier transform of the input image.
+        """
+        B, C, XYZ = self.n_batchs, self.n_coils, self.shape
+        data = data.reshape((B, 1 if self.uses_sense else C, *XYZ))
+
+        if self.smaps is not None:
+            self.smaps = self.smaps.to(data.dtype, copy=False)
+
+        samples_op = self.samples
+        if self.check_samples_shape(samples_op):
+            samples_op = torch.transpose(samples_op, 1, 0)
+        kdata = self._tkb_op.forward(image=data, omega=samples_op, smaps=self.smaps)
```

Comment:
```suggestion
        kdata = self._tkb_op.forward(image=data, omega=self.samples.t(), smaps=self.smaps)
```

We can keep things simple here. We know that samples will always be N x D. Also we know that omega is D X N. We dont need additional shape testing functions etc. This way, on the surface the API is the same.

Reply:
ok simpler et prettier